components/script/dom/gamepad.rs

```diff
@@ -66,6 +67,57 @@ impl Gamepad {
             hand: hand,
         }
     }
+
+    pub fn new(global: &GlobalScope, gamepad_id: u32, id: String) -> DomRoot<Gamepad> {
+        let gamepad = Self::new_with_proto(global, gamepad_id, id);
+        gamepad.init_axes();
+        gamepad
+    }
+
+    fn new_with_proto(global: &GlobalScope, gamepad_id: u32, id: String) -> DomRoot<Gamepad> {
+        // https://www.w3.org/TR/gamepad/#dfn-initializing-buttons
+        // https://www.w3.org/TR/gamepad/#fingerprinting-mitigation
+        // Initialize the number of buttons in the "standard" gamepad mapping.
+        // The spec says UAs *may* do this for fingerprint mitigation, and it also
+        // happens to simplify implementation
+        let standard_buttons = &[
+            GamepadButton::new(global, false, false), // Bottom button in right cluster
+            GamepadButton::new(global, false, false), // Right button in right cluster
+            GamepadButton::new(global, false, false), // Left button in right cluster
+            GamepadButton::new(global, false, false), // Top button in right cluster
+            GamepadButton::new(global, false, false), // Top left front button
+            GamepadButton::new(global, false, false), // Top right front button
+            GamepadButton::new(global, false, false), // Bottom left front button
+            GamepadButton::new(global, false, false), // Bottom right front button
+            GamepadButton::new(global, false, false), // Left button in center cluster
+            GamepadButton::new(global, false, false), // Right button in center cluster
+            GamepadButton::new(global, false, false), // Left stick pressed button
+            GamepadButton::new(global, false, false), // Right stick pressed button
+            GamepadButton::new(global, false, false), // Top button in left cluster
+            GamepadButton::new(global, false, false), // Bottom button in left cluster
+            GamepadButton::new(global, false, false), // Left button in left cluster
+            GamepadButton::new(global, false, false), // Right button in left cluster
+            GamepadButton::new(global, false, false), // Center button in center cluster
+        ];
+        rooted_vec!(let buttons <- standard_buttons.iter().map(|button| DomRoot::from_ref(&**button)));
+        let button_list = GamepadButtonList::new(global, buttons.r());
```

Comment:
It would be good to:
1. move the #dfn-initializing-buttons logic to a separate method, like you did with init axes
2. Add comments about TODOs

Reply:
Done